synapse/handlers/federation_event.py

```diff
@@ -1939,3 +1949,33 @@ def _sanity_check_event(self, ev: EventBase) -> None:
                 len(ev.auth_event_ids()),
             )
             raise SynapseError(HTTPStatus.BAD_REQUEST, "Too many auth_events")
+
+    def strip_unsigned_values(self, pdu: EventBase, event_dict: Dict) -> EventBase:
+        """
+        Strip any unsigned values unless specifically allowed, as defined by the whitelist.
+
+        pdu: the event to strip values from
+        event_dict: a dict of values derived from the pdu
+        """
+        unsigned = event_dict.get("unsigned")
+
+        if pdu.type == EventTypes.Member:
+            whitelist = ["knock_room_state", "invite_room_state", "age"]
+        else:
+            whitelist = ["age"]
+
+        filtered_unsigned = {}
+
+        # Unsigned should never be None by the time we are here but mypy doesn't know that
```

Comment:
Is that because of the checks preceding each call of `strip_unsigned_values`? It might be reasonable to just use `unsigned = event_dict["unsigned"]` and avoid the assert in this case.